### ISSUE TITLE
Dashboard not loading +Error Perl execution failed

### ISSUE THREAD

**Issue by flw55** [webmin/webmin#1156]
Hello, my webmin is not working properly.

I can't view Dashboard, it's not loading. And Favorites shows error.

Please see this: 
![fav](https://user-images.githubusercontent.com/46347688/70898585-3384a100-1ff5-11ea-808a-e2741209e57f.png)

Webmin 1.930

It worked 1 week ago, I don't know what happened.

**Comment by rostovtsev** [comment]
Try deleting `/etc/webmin/authentic-theme/favorites-(username).json` file, and reloading the page (in browser).

In case the file for favorites was manually edited, and the syntax was broken, it might end-up with a load error.

**Comment by flw55** [comment]
It wasn't manually edited but it worked! Thank you.

**Comment by rostovtsev** [comment]
I have made more tests, and there seems to present an XSS issue with theme _Favorites_, that can be reproduced in some rare cases. I will address it on the commit tomorrow.

Thanks for reporting.

**Comment by rostovtsev** [comment]
Okay, I have made multiple important fixes to theme favorites management, removing old school escaping and double encoding.

Besides, now, even though the JSON file with favorites corrupted, the theme load will not be stopped/collapsed and just continue the load. It's easy to break JSON file when editing manually, so it's surely a good/safe guard to have.

I have discovered more XSS when saving a file with File Manager but this I will fix tomorrow.

Thank you for reporting! :wink:
